content/hardware/10.mega/boards/giga-r1-wifi/tutorials/giga-debugging/giga-debugging.md

```diff
@@ -0,0 +1,184 @@
+
+This article covers the basic steps for debugging both cores on the GIGA R1 WiFi using the SWD pins and a J-Link® debugger.
+
+In addition to the necessary hardware we also need some software allowing us to set breakpoints in our code. A breakpoint is an intentional stopping or pausing place at a specific point in the code, allowing you to read values at that exact point. In this case, we will be using Ozone which is a software developed by Segger®. It’s a graphical debugger for embedded applications and we can use it to set breakpoints, read out memory addresses, and read the value of any variable at a specific point in time.
+
+## Goals
+
+The goals of this tutorial are:
+
+- learn to debug both cores on the GIGA R1 WiFi.
+- learn how to connect the GIGA R1 WiFi to the Segger J-Link debugger.
+- learn how to use the Ozone debugger software to debug an Arduino sketch.
+
+## Hardware & Software Needed
+
+- [Arduino IDE](https://www.arduino.cc/en/main/software)
+- [Arduino GIGA R1 WiFi](https://store.arduino.cc/products/giga-r1-wifi)
+- [GIGA R1 WiFi Board Package](/tutorials/giga-r1-wifi/giga-getting-started/)
+- [Segger J-Link](https://www.segger.com/products/debug-probes/j-link/)
+- [Ozone](https://www.segger.com/products/development-tools/ozone-j-link-debugger/)
+
+## Debugging
+
+Debugging your Arduino project allows you to dive deep into your code and troubleshoot as well as analyze the code execution. J-Link is a popular debugging tool that provides a direct connection between your computer and the Arduino board. By connecting it you can gain full access to the microcontroller's internal registers, memory, and variables. This is especially helpful when working on more complex projects where understanding the code execution flow is crucial. With J-Link and Ozone, you can step through the code line by line, allowing you to analyze why your code might break at a specific point.
+
+## Connection
+
+Connecting the J-link to your GIGA R1 WiFi is super easy because there are special pins for debugging labeled as **SWD** pins.
+
+![SWD Pins](./assets/jtag_pins.png)
+
+Your J-link should come with a cable included and if you look closer you will see that one side is marked with red, indicating the orientation of the cable. The red side of the cable should be on the same side as the VCC pin on the SWD pad. To know the exact orientation of your cable be sure to check the respective documentation or datasheet of your debugger but you can start with the orientation as shown below and if it doesn't work you can try placing it differently until you get a working connection. Once we start using the debugger software in the following steps you will know if the cable is placed correctly.
+
```

Comment:
Missing image

Reply:
I added a link to Segger's documentation instead.